### DISCUSSION TITLE
Correlation coeffecient for sdmTMB?

### DISCUSSION THREAD

**Discussion by leoohyama** [sdmTMB/sdmTMB#145] [Q&A]
Hello,

I was wondering if there are any ways to calculate a correlation coefficient from an sdmtmb model? I've tried some popular R-square functions (e.g. r.squaredglmm() from the package MuMin) but none of these functions have been updated to handle model class of sdmTMB.  The models I'm running are simple spatial glms with no extra random effects.

**Comment by ericward-noaa** [comment]
What kind of distribution are you assuming for the response? I think if your data are ~ normal, you could just calculate this yourself after calling the predict() function on the sdmTMB model object. There's not a built in function for this yet though -- perhaps we should though. Here's a thread on the development of pseudo-R squared for glmmTMB

https://github.com/glmmTMB/glmmTMB/issues/169

There the authors extended the performance package to handle glmmTMB objects

**Reply by leoohyama** [reply]
I'm implementing a Gamma distribution with log link. 
I have tried the performance package but to no success. 
I am wondering if R2 are necessary when reporting models for simply inference rather than predictions.

**Reply by ericward-noaa** [reply]
Great -- thanks. I think performance won't work until we add sdmTMB objects on the backend. 

But to the larger point, I wouldn't worry about reporting R2 if you're just doing inference (and not trying to measure predictive accuracy). The other thing you could do with cross validation in sdmTMB_cv() is calculate the log-score and compare that between models (a challenge with R2 across models with different complexity is that you have to calculated the adjusted R2 version)

**Reply by leoohyama** [reply]
Got it! 
Thanks for the thoughts on the larger point!

**Comment by seananderson** [comment]
I worked out the Nakagawa/Schielzeth/Johnson R^2 here: https://github.com/pbs-assess/sdmTMB/blob/main/scratch/r2.R I think partially following what's done for glmmTMB here https://github.com/glmmTMB/glmmTMB/blob/master/glmmTMB/inst/misc/rsqglmm.R

But I stopped at the Gaussian family. It would be good to expand that and some point and maybe incorporate it into the package.

**Reply by leoohyama** [reply]
Thanks for the link. I will check it out!
